core/src/main/java/com/linecorp/armeria/common/util/AbstractOptions.java

```diff
@@ -19,31 +19,60 @@
 
 import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.IdentityHashMap;
+import java.util.Iterator;
 import java.util.Map;
 import java.util.NoSuchElementException;
 
 import javax.annotation.Nullable;
 
+import com.google.common.collect.Iterators;
+
 /**
  * A set of configuration options and their respective values.
  *
  * @see AbstractOption
  * @see AbstractOptionValue
  */
-public abstract class AbstractOptions {
+public abstract class AbstractOptions<T extends AbstractOptionValue<?, ?>> implements Iterable<T> {
+
+    protected static <V> V get(AbstractOptions<?> first, AbstractOptions<?> second, AbstractOption<V> option) {
```

Comment:
- Could we make sure `first` and `second` have the same type parameter?
- Missing Javadoc